src/Zadatak2/PogadjanjeSlucajnogBroja.java

```diff
@@ -0,0 +1,107 @@
+package zadatak2;
+
+import java.util.InputMismatchException;
+import java.util.Random;
+import java.util.Scanner;
+
+/**
+ * 
+ * @author Goran.Kukolj
+ *
+ *         ZADATAK2: Napisati program za pogadjanje slucajno generisanog broja u
+ *         opsegu od 1 do 50. Program treba da radi na sledeci nacin: Korisnik
+ *         preko standardnog ulaza unosi jedan ceo broj. Broj koji treba
+ *         pogoditi generisati �random� u opsegu od 1 do 50. Vrteti se u petlji
+ *         dok ne pogodimo broj, ili budemo barem blizu (ako je razlika izmedju
+ *         brojeva 5 ili manja). Taj deo sa proverom pogotka izdvojiti u zasebnu
+ *         metodu, kojoj se kao argument prosledjuje broj koji tipujemo, a
+ *         povratna vrednost metode je 0 � ako smo pogodili, 1 � ako smo bili
+ *         blizu, 2 � ako smo promasili. U zavisnosti od rezultata pogadjanja
+ *         ispisati i odgovarajucu poruku. Nakon odigrane partije proveriti da
+ *         li korisnik zeli da prekine ili da nastavi igru.
+ */
+public class PogadjanjeSlucajnogBroja {
+
+	public static void main(String[] args) {
+
+		// kreiramo random objekat da dobijemo random zamisljeni broj
+		Random random = new Random();
+		System.out.println("Dobrodosli u igru pogadjanja");
+		boolean novaIgra = true;
+
+		// pocinje igra u petlji i ponavlja se koliko hocemo puta
+		while (novaIgra) {
+
+			// kreiramo slucajan broj izmedju 1 i 50
+			int slucajanBroj = random.nextInt(49) + 1;
```

Comment:
Ovde nisu potrebne dve petlje, vec je dovoljna jedna. Ovo sto ide pre do-while petlje a trenutno je while, moze da se stavi u do-while. Ovo generisanje random broja svakako treba da ide u do-while.

Reply:
Pitanje, ako stavim generisanje random broja u do-while, on ce u svakom prolazu generisati novi broj,jel to treba tako?

Reply:
Ja sam kod mene ispravio tako i radi. Treba voditi racuna o kompleksnosti metoda zbog brzine izvrsavanja i i optercenja memorije, znaci ako se moze resiti jednom petljom treba tako uraditi